x-pack/plugins/discover_log_explorer/public/plugin.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CoreStart, Plugin } from '@kbn/core/public';
+import { LOG_EXPLORER_PROFILE_ID } from '../common/constants';
+import { createLazyCustomDatasetSelector } from './customizations';
+import {
+  DiscoverLogExplorerPluginSetup,
+  DiscoverLogExplorerPluginStart,
+  DiscoverLogExplorerStartDeps,
+} from './types';
+
+export class DiscoverLogExplorerPlugin
+  implements Plugin<DiscoverLogExplorerPluginSetup, DiscoverLogExplorerPluginStart>
+{
+  public setup() {}
+
+  public start(core: CoreStart, plugins: DiscoverLogExplorerStartDeps) {
+    const { discover } = plugins;
+
+    discover.customize(LOG_EXPLORER_PROFILE_ID, async ({ customizations, stateContainer }) => {
```

Comment:
The body of this function could grow quite a lot over time and make the plugin class harder to read at a glance. I wonder if we should factor out the async function into `customizations/log_explorer_profile.ts` or so?

Reply:
Yes it makes sense, I'm not a fan too of having much more logic in the plugin, and even if this is not too long at the current state it is a good idea to factor it out.